aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -193,6 +193,14 @@ public static <T extends SdkClientBuilder> void configureEndpoint(T builder, Str
     }
   }
 
+  /**
+   * Build an S3Configuration object from two boolean flags, pathStyleAccess and
+   * s3UseArnRegionEnabled
+   *
+   * @deprecated Not for public use. To build an S3Configuration object, use
+   *     S3Configuration.builder() directly.
```

Comment:
This should include a note like "will be removed in 2.0.0".

Reply:
Added